modules/tinymce/src/themes/silver/test/ts/browser/editor/core/UndoRedoUiTest.ts

```diff
@@ -0,0 +1,59 @@
+import { UiFinder, Waiter } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { SugarBody } from '@ephox/sugar';
+import { TinyHooks, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+
+describe('browser.tinymce.themes.silver.editor.core.UndoRedoUiTest', () => {
+  const hook = TinyHooks.bddSetupLight<Editor>({
+    base_url: '/project/tinymce/js/tinymce',
+    toolbar: 'redo undo',
+    menubar: true
+  }, []);
+
+  const assertToolbarButtonDisabled = (title: string, state: boolean = true) =>
+    UiFinder.exists(SugarBody.body(), `button[title="${title}"][aria-disabled="${state}"]`);
+
+  const pAssertMenuItemDisabled = (item: string, state: boolean = true) =>
+    Waiter.pTryUntil('Wait for a specific menu to open', () => UiFinder.exists(SugarBody.body(), `[role="menuitem"][aria-disabled=${state}"]:contains("${item}")`));
+
+  it('TINY-8101: Undo/redo should function as expected in normal editor mode', () => {
```

Comment:
This is really 2 separate test cases, so can we please break it up to match that? 1) for checking they are always disabled with a clean state and then 2) for checking if they are enabled/disabled when you have dirty state.